yt/yt/server/lib/state_checker/state_checker.cpp

```diff
@@ -0,0 +1,88 @@
+#include "private.h"
+#include "state_checker.h"
+
+#include <yt/yt/core/logging/log.h>
+
+#include <yt/yt/core/concurrency/periodic_executor.h>
+
+namespace NYT::NStateChecker {
+
+using namespace NApi;
+using namespace NConcurrency;
+using namespace NLogging;
+using namespace NYson;
+using namespace NYTree;
+
+////////////////////////////////////////////////////////////////////////////////
+
+TStateChecker::TStateChecker(IInvokerPtr invoker, IClientPtr nativeClient, TYPath instancePath, TDuration stateCheckPeriod)
+    : Logger(StateCheckerLogger())
+    , Invoker_(std::move(invoker))
+    , NativeClient_(std::move(nativeClient))
+    , InstancePath_(std::move(instancePath))
+{
+    Banned_.store(false);
+    StateCheckerExecutor_ = New<TPeriodicExecutor>(
+        Invoker_,
+        BIND(&TStateChecker::DoCheckState, MakeStrong(this)),
+        stateCheckPeriod);
+}
+
+void TStateChecker::Start()
+{
+    StateCheckerExecutor_->Start();
+}
+
+void TStateChecker::SetPeriod(TDuration stateCheckPeriod)
+{
+    StateCheckerExecutor_->SetPeriod(stateCheckPeriod);
+}
+
+bool TStateChecker::IsComponentBanned() const
+{
+    return Banned_.load();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+void TStateChecker::DoCheckState()
+{
+    VERIFY_INVOKER_AFFINITY(Invoker_);
+
+    YT_LOG_DEBUG("Started checking component state");
+    auto logFinally = Finally([&] {
+        YT_LOG_DEBUG("Finished checking component state (Banned: %v)", Banned_.load());
+    });
+
+    auto options = TGetNodeOptions{
+        .Attributes = TAttributeFilter({BannedAttributeName}),
+    };
+
+    try {
+        auto yson = WaitFor(NativeClient_->GetNode(InstancePath_, options))
+            .ValueOrThrow();
+        auto instance = ConvertToNode(yson);
+
+        auto banned = false;
+        if (instance->Attributes().Contains(BannedAttributeName)) {
+            banned = instance->Attributes().Get<bool>(BannedAttributeName);
+        }
+
+        auto previousBanned = Banned_.exchange(banned);
+        if (previousBanned != banned) {
+            auto orchidBannedPath = InstancePath_ + "/orchid/@banned";
```

Comment:
Orchid usually works as a virtual YTree for diagnostics.
What you've done here with direct sets into cypress is really confusing.

You can create a pluggable orchid service in StateChecker, that will be incorporated into ochid on yql agent and QT.

Look for example at methods GetOrchidService.